### DISCUSSION TITLE
Passing new updated data but rendering with initial data sent to the tree

### DISCUSSION THREAD

**Discussion by Arjav97** [dowjones/react-dropdown-tree-select#522] [Q&A]
What i am doing is in the onChange function i am setting the selected nodes in the state using useState and then rendering the selected nodes on right of the tree where each selected label is displayed with cross symbol and onClick of that cross button the node is deleted from the list and also updating the data by setting checked to false of that particular node and then when I am passing that deep cloned updated data to child component then all the checkboxes are unchecked as it is set with initial data.

I dont know the reason why it is happening since I am rendering child with HOC wherein I am using shouldComponent and willReceiveProps to identify the updated data that is sent as props is different from the previous data.

Need of any suggestions as if this setting up with new updated data cannot be done then I would have to find any other route of doing this dropdown tree

**Comment by mrchief** [comment]
Your issue sounds similar to https://github.com/dowjones/react-dropdown-tree-select/issues/313#issuecomment-564676498

Can you create a codesandbox that reproduces your issue?

**Comment by Arjav97** [comment]
I have reproduced the issue in codebox and the link is 

https://codesandbox.io/s/rough-worker-t3i5b?file=/src/App.js

See how the action on list of nodes on pressing cross symbol deletes or refreshes the react dropdown tree with initial data since still parent rerender HOC is used and component lifecycle methods are used and also the data sent is deep copy so no mutating data directly sending everytime new object of data

**Comment by Arjav97** [comment]
Hi Hrusikesh I have attached the codesandbox link if you can look through
let me know if there is any way around to solver it.

**Comment by mrchief** [comment]
I'm getting this error when deleting 
![image](https://user-images.githubusercontent.com/781818/135354962-f7d9c87d-da01-464b-9740-4e318609e476.png)

**Comment by Arjav97** [comment]
I have attached the updated link since I attached the wrong one...

The problem in this is that selecting on tree works well and deletion from the right portion of list also works but when I am doing unchecking directly on the tree I start facing uncertain results if you can go through this would be great.

[https://codesandbox.io/s/rough-worker-t3i5b?file=/src/App.js](url)

**Comment by Arjav97** [comment]
I guess this is the correct link !!! 
https://codesandbox.io/s/rough-worker-t3i5b?file=/src/App.js

The problem is when I am selecting and deselecting from the list on the right side with cross symbol then updated state is reflected on tree but when I am unchecking on tree itself it shows change in selected nodes but it is not updated/reflected on the tree itself!!!

**Comment by mrchief** [comment]
I made a few tweaks and you can see the approach here https://codesandbox.io/s/focused-field-4qt17?file=/src/App.js

You'll need to handle selection/deselection of parent nodes but this is to show you the overall idea.
